### PR TITLE
docs: expose zone config in schema doc

### PR DESCRIPTION
**Pull request by zmstone**
Fixes https://emqx.atlassian.net/browse/EMQX-12162
Release version: v/e5.7.0

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx/src/emqx_schema.erl [hunk 1]
```diff
@@ -1883,6 +1876,22 @@ base_listener(Bind) ->
             )}
     ] ++ emqx_limiter_schema:short_paths_fields().
 
+%% @hidden Starting from 5.7, listeers.{TYPE}.{NAME}.zone is no longer hidden
```
Comment (ieQu1) [inline apps/emqx/src/emqx_schema.erl:1879]:
```suggestion
%% @hidden Starting from 5.7, listeners.{TYPE}.{NAME}.zone is no longer hidden
```

File: apps/emqx/src/emqx_schema.erl [hunk 2]
```diff
@@ -1883,6 +1876,22 @@ base_listener(Bind) ->
...
+%% However, the root key 'zones' is still hidden because the fields' schema
+%% just repeat other root fied's schema, which makes the dumped schema doc
```
Comment (ieQu1) [inline apps/emqx/src/emqx_schema.erl:1881]:
```suggestion
%% just repeat other root field's schema, which makes the dumped schema doc
```

**Review by ieQu1 — APPROVED** [review]

File: apps/emqx/src/emqx_schema.erl [hunk 3]
```diff
@@ -1883,6 +1876,22 @@ base_listener(Bind) ->
             )}
     ] ++ emqx_limiter_schema:short_paths_fields().
 
+%% @hidden Starting from 5.7, listenrs.{TYPE}.{NAME}.zone is no longer hidden
```
[the remark this answers is not in the document]
Comment (zmstone) [reply apps/emqx/src/emqx_schema.erl:1879]: will fix in some other pr.

**Comment by yanzhiemq** [comment]
[TestExecution](https://emqx.atlassian.net/browse/QA-2151)

**Pull request merged**